전남대 FE_임지환 3주차 과제

dlawlghks opened the pull request:
안녕하세요 멘토님!
이번 주 부터 과제 제출 방식이 변경되어 step1 먼저 제출합니다!!
step1을 진행해보았는데 혹시 잘못된 점이나 부족한 점 있는지 봐주시면 바로 고치겠습니다!!

감사합니다.

dlawlghks committed:
docs: 초기 README 추가

dlawlghks committed:
chore: package.json과 package-lock.json에 의존성 추가

dlawlghks committed:
feat: src/api/api.ts 파일 추가 및 API 요청 함수 구현

dlawlghks committed:
feat: useCurrentTheme 커스텀 훅 추가

dlawlghks committed:
style: Filter.tsx 스타일 수정

dlawlghks committed:
feat: GoodsRankingSection 컴포넌트에 로딩 및 에러 상태 추가

dlawlghks committed:
feat: ThemeCategorySection 컴포넌트에 동적 테마 데이터 추가 및 로딩/에러 처리 추가

dlawlghks committed:
feat: ThemeGoodsSection 컴포넌트에 동적 상품 데이터 및 로딩/에러 처리 추가

dlawlghks committed:
feat: ThemeHeroSection 컴포넌트에 useCurrentTheme 훅 추가 및 동적 데이터 렌더링

dlawlghks committed:
fix: import 수정

dlawlghks committed:
fix: ThemePage import 수정 및 useCurrentTheme 훅 적용 수정

dlawlghks committed:
feat: API 응답 타입 정의 추가 (GoodsResponse, ThemesResponse, RankingResponse, ProductsResponse)

dlawlghks committed:
feat: ThemeMockData에 필드 추가

tlsehdfl commented on src/api/hooks/useCurrentTheme.tsx:
```diff
@@ -0,0 +1,26 @@
+};
+
+export const useCurrentTheme = ({ themeKey }: UseCurrentThemeProps) => {
+  const [isRender, setIsRender] = useState(false);
```
해당 state 는 왜 필요한가요 ?

tlsehdfl commented on src/components/features/Home/GoodsRankingSection/index.tsx:
```diff
@@ -15,19 +15,43 @@ export const GoodsRankingSection = () => {
-  // GoodsMockData를 21번 반복 생성
+  const [goodsList, setGoodsList] = useState<GoodsData[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<string | null>(null);
```
`isError` 면 boolean 을 나타낼거 같은데 message 로 지정되어 있네요.

```suggestion
  const [errorMessage, setErrorMessage] = useState<string>('');
```

tlsehdfl commented on src/components/features/Home/GoodsRankingSection/index.tsx:
```diff
@@ -15,19 +15,43 @@ export const GoodsRankingSection = () => {
+  useEffect(() => {
+    const fetchRankingProducts = async () => {
+      setIsLoading(true);
+      setIsError(null);
```
```suggestion
```

tlsehdfl commented on src/components/features/Home/ThemeCategorySection/index.tsx:
```diff
@@ -1,14 +1,44 @@
+        setLoading(false);
+      } catch (err) {
+        setError('Failed to fetch themes');
+        setLoading(false);
```
`catch` 가 아니라 `finally` 절로 가야 할 거 같습니다.

tlsehdfl replied:
성공이든 에러든 완료시 에러 상태가 제거되어야 하기 때문입니다.

tlsehdfl commented on src/components/features/Theme/ThemeGoodsSection/index.tsx:
```diff
@@ -1,16 +1,52 @@
+
+  useEffect(() => {
+    const fetchThemeProducts = async () => {
+      if (!themeKey) {
```
`themeKey` 가 없다면 / 으로 navigation 하는 건 `fetchThemeProducts` 밖에 있어야 할 거 같습니다

tlsehdfl commented on src/components/features/Theme/ThemeHeroSection/index.tsx:
```diff
@@ -1,16 +1,18 @@
-  const currentTheme = getCurrentTheme(themeKey, ThemeMockList);
+  const { isRender, currentTheme } = useCurrentTheme({ themeKey });
+
+  if (!isRender) return null;
```
`isRender` 상태는 불필요합니다. `currentTheme` 이 없으면 null 로 리턴하면 될 거 같습니다.

tlsehdfl commented on src/pages/Theme/index.tsx:
```diff
@@ -1,13 +1,15 @@
-  const currentTheme = getCurrentTheme(themeKey, ThemeMockList);
+  const { isRender, currentTheme } = useCurrentTheme({ themeKey });
+
+  if (!isRender) return null;
```
`isRender` 상태는 불필요합니다. `currentTheme` 이 없으면 null 로 리턴하면 될 거 같습니다.

dlawlghks committed:
refactor: 로딩 상태 관리를 위해 isRender를 isLoading으로 교체

- useCurrentTheme 훅에서 isRender를 isLoading으로 교체
- ThemeHeroSection 및 ThemePage 컴포넌트에서 isLoading을 사용하도록 업데이트

dlawlghks committed:
refactor: 변경 사항 반영 및 상태 변수 명확화

dlawlghks committed:
fix: ThemeCategorySection에서 setLoading을 finally 블록으로 이동

dlawlghks committed:
fix: themeKey가 없을 경우 navigate('/') 호출 위치 수정

dlawlghks committed:
refactor: 불필요한 코드 제거

dlawlghks committed:
refactor: isRender 상태 제거 및 currentTheme 존재 여부로 처리

dlawlghks committed:
refactor: isRender 상태 제거 및 currentTheme 존재 여부로 처리

dlawlghks commented:
수정 완료했습니다!
감사합니다^__^

dlawlghks committed:
feat: react-spinners 패키지 추가

dlawlghks committed:
docs: README 업데이트

dlawlghks committed:
feat: 상품 랭킹 섹션 로딩 스피너 추가 및 오류 처리 개선

dlawlghks committed:
feat: 테마 상품 섹션에 로딩 스피너 및 오류 처리 개선

dlawlghks commented:
멘토님 안녕하세요!
추가로 STEP2도 진행해보았습니다. 
수정사항이 있다면 바로 고치겠습니다.

항상 피드백해주셔서 감사합니다!

dlawlghks committed:
feat: 페이지네이션 지원 추가

dlawlghks committed:
feat: useCurrentTheme 추가

dlawlghks committed:
chore: useCurrentTheme 파일 확장자 .tsx에서 .ts로 변경

dlawlghks committed:
feat: useInfiniteThemeProducts 훅 추가하여 무한 스크롤 기능 구현

dlawlghks committed:
feat: instance.ts 추가

dlawlghks committed:
feat: QueryClientProvider로 애플리케이션 감싸기

dlawlghks committed:
refactor: ThemeGoodsSection 컴포넌트 리팩토링 및 useInfiniteQuery 사용하여 무한 스크롤 기능 추가

dlawlghks committed:
refactor: App 컴포넌트 경로 수정 및 ReactDOM.createRoot 타입 단언 추가

dlawlghks committed:
refactor: HomePage 컴포넌트에 React.FC 타입 명시

dlawlghks committed:
docs: README 수정

dlawlghks commented:
멘토님 3주차 과제를 모두 마무리해보았습니다!
감사합니다.
